src/Meta/MetaFactory.cpp

```diff
@@ -357,7 +357,50 @@ void MetaFactory::createFromMethod(const clang::ObjCMethodDecl& method, MethodMe
         }
     }
 
-    methodMeta.setFlags(MetaFlags::MethodIsInitializer, method.getMethodFamily() == clang::ObjCMethodFamily::OMF_init); // set MethodIsInitializer flag
+    bool isInitializer = (method.getMethodFamily() == clang::ObjCMethodFamily::OMF_init);
+    methodMeta.setFlags(MetaFlags::MethodIsInitializer, isInitializer); // set MethodIsInitializer flag
+    if (isInitializer) {
+        assert(methodMeta.getSelector().find("init", 0) == 0);
```

Comment:
Theoretically it is possible an arbitrary method to belong to the `init` family if it is decorated with clang attribute:
```
- (id) myMethod __attribute__((objc_method_family(init))) { ... }
```
or a method starting with `init` and not belonging to the `init` family:
```
- (id) initWithSomething __attribute__((objc_method_family(none))) { ... }
```
These are very rare cases and currently we don't handle them and in these cases the assert will fail in debug builds only.
Also, if we decide to consider methods, not starting with `init`, as initializers (e.g. we may begin respecting `objc_designated_initializer` clang attribute which I believe may be added to an arbitrary method) we have to change the logic introduced in this PR because it assumes that all initializers start with `init` and the failing assert will prompt us to do it.